.github/workflows/workflow-path-to-live.yml

```diff
@@ -0,0 +1,347 @@
+      push_to_ecr: true
+    secrets: inherit
+
+  api_unit_tests:
+    name: api unit tests
+    uses: ./.github/workflows/_unit-tests-api.yml
+    needs:
+      - workflow_variables
+      - docker_build_scan_push
+
+  client_unit_tests:
+    name: client unit tests
+    uses: ./.github/workflows/_unit-tests-client.yml
+    needs:
+      - workflow_variables
+      - docker_build_scan_push
+
+  codecov:
+    name: upload to codecov
+    uses: ./.github/workflows/_codecov.yml
+    needs:
+      - client_unit_tests
+      - api_unit_tests
+    secrets: inherit
+
+  synchronisation_unit_tests:
+    name: client unit tests
+    uses: ./.github/workflows/_unit-tests-synchronise-lambda.yml
+    needs:
+      - workflow_variables
+      - docker_build_scan_push
+
+  monitoring_lambda_unit_tests:
+    name: monitoring lambda unit tests
+    uses: ./.github/workflows/_unit-tests-monitoring-lambda.yml
+    needs:
+      - workflow_variables
+
+  terraform_apply_shared_development:
+    name: shared plan terraform development
+    uses: ./.github/workflows/_run-terraform.yml
+    needs:
+      - workflow_variables
+    with:
+      workspace: development
+      terraform_path: shared
+      container_version: ${{ needs.workflow_variables.outputs.parsed_branch }}-${{ needs.workflow_variables.outputs.short_sha }}
+    secrets: inherit
+
+  terraform_apply_development:
+    name: environment apply terraform
+    uses: ./.github/workflows/_run-terraform.yml
+    needs:
+      - terraform_apply_shared_development
+      - docker_build_scan_push
+      - monitoring_lambda_unit_tests
+      - workflow_variables
+    with:
+      workspace: development
+      terraform_path: environment
+      apply: true
+      container_version: ${{ needs.workflow_variables.outputs.parsed_branch }}-${{ needs.workflow_variables.outputs.short_sha }}
+    secrets: inherit
+
+  terraform_apply_shared_preproduction:
+    name: shared plan terraform development
+    uses: ./.github/workflows/_run-terraform.yml
+    needs:
+      - workflow_variables
+      - terraform_apply_development
+    with:
+      workspace: preproduction
+      terraform_path: shared
+      container_version: ${{ needs.workflow_variables.outputs.parsed_branch }}-${{ needs.workflow_variables.outputs.short_sha }}
+    secrets: inherit
+
+  terraform_apply_integration:
+    name: environment apply terraform
+    uses: ./.github/workflows/_run-terraform.yml
+    needs:
+      - terraform_apply_shared_preproduction
+      - workflow_variables
+    with:
+      workspace: integration
+      terraform_path: environment
+      apply: true
+      container_version: ${{ needs.workflow_variables.outputs.parsed_branch }}-${{ needs.workflow_variables.outputs.short_sha }}
+    secrets: inherit
+
+  scale_services_up:
+    name: scale up services
+    uses: ./.github/workflows/_scale-services.yml
+    needs:
+      - workflow_variables
+      - terraform_apply_integration
+    with:
+      replicas: 6
```

Comment:
Yeah, I don't think much has changed since i set this up and I did test it out with different numbers and this was the sweet spot. I could have a go at rejigging this at some point.